src/Controls/src/SourceGen/IndentedTextWriterExtensions.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.CodeDom.Compiler;
+using System.Linq;
+
+namespace Microsoft.Maui.Controls.SourceGen;
+
+static class IndentedTextWriterExtensions
+{
+	public static void Append(this IndentedTextWriter writer, IndentedTextWriter other, bool noTabs = false)
+	{
+		other.Flush();
+		var lines = other.InnerWriter.ToString().Split([other.InnerWriter.NewLine], StringSplitOptions.None).ToArray();
```

Comment:
Why is the `.ToArray()` needed here? Does `Split()` already return a `string[]`?

If this is called a lot, a future PR, you might consider interating over the lines using `Span<char>` on `InnerWriter.ToString()` and it could avoid creating a `string[]`.